### PR TITLE
Add ux configuration to charts. Allow callers to override default UX.

### PR DESCRIPTION
**Pull request by parc-jason**
Signed-off-by: Jason Sherman <jsherman@parcsystems.ca>

Relates to https://github.com/hyperledger-labs/business-partner-agent/pull/477. This supports the configuration changes in that PR.

### CODE REVIEW

File: charts/bpa/templates/_helpers.tpl [hunk 1]
```diff
@@ -272,12 +272,18 @@ Create environment variables for database configuration.
 Return JAVA_OPTS -Dmicronaut.config.files
 */}}
 {{- define "bpa.config.files" -}}
-{{- if and .Values.keycloak.enabled .Values.schemas.enabled -}}
+{{- if and .Values.keycloak.enabled .Values.schemas.enabled .Values.ux.enabled -}}
```
Comment (frank-bee) [inline charts/bpa/templates/_helpers.tpl:275]: Maybe you could create a list `$configfiles` and add the filenames based on the enablement
`$configfiles | join ","`

Comment (frank-bee) [reply charts/bpa/templates/_helpers.tpl:275]: see also https://helm.sh/docs/chart_template_guide/function_list/#lists-and-list-functions

File: charts/bpa/templates/bpa_configmap_ux.yaml [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+  name: {{ include "bpa.fullname" . }}-ux
+  labels:
+    {{- include "bpa.labels" . | nindent 4 }}
+data:
+  ux.yml: |-
+    bpa:
+      ux:
+{{- if .Values.ux.config }}
+{{ toYaml .Values.ux.config | indent 8 }}
+{{ toYaml .Values.ux.config | indent 8 }}
```
Comment (frank-bee) [inline charts/bpa/templates/bpa_configmap_ux.yaml:14]: is this line duplicated?

File: charts/bpa/values.schema.json [hunk 3]
```diff
@@ -470,7 +470,15 @@
                             "type": "boolean"
                         }
                     }
+                },
+                "ux": {
+                    "type": "string",
+                    "enum": ["bosch", "bcgov"],
+                    "description": "The UX look and feel configuration",
```
Comment (frank-bee) [inline charts/bpa/values.schema.json:477]: Currently I used `description` and `title` only if I need it in the form.
Reason: schema is not seen by other users , the look more into the values file and the Readme.md which is generated out of the values file. 
Unfortunately currently in helm there is values file and schema file.

File: charts/bpa/templates/_helpers.tpl (hunk 1 above)
Comment (frank-bee) [reply charts/bpa/templates/_helpers.tpl:275]: (I think there is also some cases missing)

File: charts/bpa/values.yaml [hunk 4]
```diff
@@ -16,6 +16,9 @@ global:
   # -- The used ledger. Will be used for default values. Any of: bosch-test, idu.
   ledger: bosch-test
 
+  # -- The ux configuration. Will be used for default values. Any of: bosch, bcgov.
+  ux: bosch
```
Comment (frank-bee) [inline charts/bpa/values.yaml:20]: MAybe you can add this "default config"  under the ux path in line 161 ?

**Review by frank-bee — CHANGES_REQUESTED** [review]
Great stuff!!
The most of my comments are just optional hints. But the construct near line 275 might miss something.

File: charts/bpa/templates/_helpers.tpl (hunk 1 above)
Comment (parc-jason) [reply charts/bpa/templates/_helpers.tpl:275]: that sounds good, I will take a look.

File: charts/bpa/templates/bpa_configmap_ux.yaml (hunk 2 above)
Comment (parc-jason) [reply charts/bpa/templates/bpa_configmap_ux.yaml:14]: ha, maybe, I'll take a look.

File: charts/bpa/templates/_helpers.tpl (hunk 1 above)
Comment (parc-jason) [reply charts/bpa/templates/_helpers.tpl:275]: Thanks for the pointer. I have changed it to use lists and it is much easier to read and understand.

File: charts/bpa/values.yaml (hunk 4 above)
Comment (parc-jason) [reply charts/bpa/values.yaml:20]: I wanted to leave this like the schemas where there are some preconfigured values and the option to push in a different object.

File: charts/bpa/templates/bpa_configmap_ux.yaml (hunk 2 above)
Comment (parc-jason) [reply charts/bpa/templates/bpa_configmap_ux.yaml]: removed, good find!

File: charts/bpa/values.schema.json (hunk 3 above)
Comment (parc-jason) [reply charts/bpa/values.schema.json]: removed.

File: charts/bpa/templates/_helpers.tpl (hunk 1 above)
Comment (frank-bee) [reply charts/bpa/templates/_helpers.tpl]: nice!

File: charts/bpa/values.yaml (hunk 4 above)
Comment (frank-bee) [reply charts/bpa/values.yaml]: got it

**Review by frank-bee — APPROVED** [review]

**Pull request merged**